qiskit/pulse/utils.py

```diff
@@ -16,9 +16,62 @@
 Pulse utilities.
 """
 import warnings
+
+from collections import namedtuple
+from typing import List
+
+from .exceptions import PulseError
+
 # pylint: disable=unused-argument
 
 
+Interval = namedtuple('Interval', ['start', 'stop'])
+"""Intervals [start, stop) are exclusive of the stop time."""
+
+
+def overlaps(first: Interval, second: Interval) -> bool:
+    """
+    Return True iff first and second overlap.
+
+    Note: first.stop may equal second.start, since Interval stop times are exclusive.
+    """
+    if first.start == second.start == second.stop:
+        # They fail to overlap if one of the intervals has duration 0
+        return False
+    if first.start > second.start:
+        first, second = second, first
+    return second.start < first.stop
+
+
+def insertion_index(intervals: List[Interval], new_interval: Interval, index: int = 0) -> int:
+    """
+    Using binary search on start times, return the index into `intervals` where the new interval
+    belongs, or raise an error if the new interval overlaps with any existing ones.
+
+    Args:
+        intervals: A sorted list of non-overlapping Intervals.
+        new_interval: The interval for which the index into intervals will be found.
+        index: A running tally of the index, for recursion. The user should not pass a value.
+
+    Returns:
+        The index into intervals that new_interval should be inserted to maintain a sorted list
+        of intervals.
+
+    Raises:
+        PulseError: If new_interval overlaps with the given intervals.
+    """
+    if len(intervals) == 1:
```

Comment:
Don't we need to care about empty intervals case? `if not intervals: return 0`?

Reply:
Yes, I agree with @itoko, I think this is an edge case that needs to be handled with a test.